zerver/views/messages.py

```diff
@@ -755,10 +753,8 @@ def same_realm_zephyr_user(user_profile, email):
     except ValidationError:
         return False
 
-    domain = email_to_domain(email)
-
     return user_profile.realm.is_zephyr_mirror_realm and \
-        RealmAlias.objects.filter(realm=user_profile.realm, domain=domain).exists()
+        get_realm_by_email_domain(email) == user_profile.realm
 
```

Comment:
Good catch! In general I agree, though this (and the two functions below it) are odd pieces of code for a few special cases. Let's leave this comment unresolved for now, and we can ask Tim what to do here after the rest of this is merged. Currently splitting this up into commits!